router.php

```diff
@@ -21,6 +21,18 @@ class Router
      */
     private array $mapper;
 
+    /**
+     * @var bool $isGrouping Defines when routes should be grouped.
+     */
+    private bool $isGrouping;
+
+    /**
+     * @var mixed $options An array of options for the group, including:
+     *   - "path": The common prefix path for all routes within this group. Defaults to an empty string.
+     *   - "middleware": An array of middleware functions to be applied to all routes within this group. Defaults to an empty array.
+     */
+    private mixed $options;

```

Comment:
Avoid using 'mixed' and try to specify the type. In this case would be an array. 
Would you reconsider turning this into a small object? Something like, for example:
class Options {
    public ?array $ConfigOverride = null;
    public ?string $path = null;

    public function __construct(?array $ConfigOverride = null, ?string $path = null) {
        $this->ConfigOverride = $ConfigOverride;
        $this->path = $path;
    }
}